adaptdl/adaptdl/torch/__init__.py

```diff
@@ -47,9 +58,17 @@ def init_process_group(backend):
             if response.status_code != 408:  # Timeout.
                 break
         response.raise_for_status()
-        master_addr = response.json()[0]
+        master_addr = response.json()['pod_ip_list'][0]
+        sched_version = response.json()['version']
+        trainer_version = pkg_resources.get_distribution("adaptdl").version
+        if version_check(sched_version, lib="sched") and \
+                version_check(trainer_version, lib="trainer"):
+            if trainer_version != sched_version:
+                raise Exception('The adaptdl version between trainer \
+                    and scheduler should be same')
```

Comment:
Checking for exact version match is too strict, since that requires every adaptdljob to use the exact same version of the adaptdl library. Instead the version checking should defer to a function that determines compatibility, i.e. whenever we make a breaking change, we add another condition to that function.

Reply:
Sure, checking compatibility instead of strict version checking makes much more sense. Do you have any suggestions about the design for detecting the breaking change?

Reply:
After discussion with Omkar and Richard, we agree to follow semantic versioning (https://semver.org/) to increment the version so the version checking of compatibility will be implemented in terms of the major version number.